### PR TITLE
SNOW-879621: Create tests performance artifact

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
Prepare tests performance artifact after test run to be able to compare the difference between environments if needed.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -104,11 +104,17 @@ jobs:
...
+      - name: Upload Code Coverage Report
         uses: actions/upload-artifact@v3
         with:
           name: code-coverage-report
           path: Snowflake.Data.Tests\${{ matrix.dotnet }}_${{ matrix.cloud_env }}_coverage.xml
+      - name: Upload Test Performance Report
+        uses: actions/upload-artifact@v3
+        with:
+          name: tests-performance
+          path: Snowflake.Data.Tests\windows_${{ matrix.dotnet }}_${{ matrix.cloud_env }}_performance.csv
```
Comment (sfc-gh-pfus) [inline .github/workflows/main.yml:117]: Why only windows?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -195,3 +207,9 @@ jobs:
           dotnet minicover uninstrument
         env:
           snowflake_cloud_env: ${{ matrix.cloud_env }}
+          net_version: ${{ matrix.dotnet }}
+      - name: Upload Test Performance Report
+        uses: actions/upload-artifact@v3
+        with:
+          name: tests-performance
+          path: Snowflake.Data.Tests\windows_${{ matrix.dotnet }}_${{ matrix.cloud_env }}_performance.csv
```
Comment (sfc-gh-pfus) [inline .github/workflows/main.yml:215]: Same?

File: Snowflake.Data.Tests/SFBaseTest.cs [hunk 3]
```diff
@@ -116,33 +138,69 @@ public void Setup()
...
+            {
+                os = "linux";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                os = "macos";
+            }
+            
+            var resultText = "test;time\n";
+            foreach (var test in s_testPerformance)
```
Comment (sfc-gh-pfus) [inline Snowflake.Data.Tests/SFBaseTest.cs:183]: Can we rewrite it to something like `s_testPerformance.map(test => $"{test.Key};{time...}").join("\n")`?

File: .github/workflows/main.yml (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply .github/workflows/main.yml:117]: Thanks for finding this :) Fixed.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -155,6 +161,12 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply .github/workflows/main.yml]: Fixed

File: .github/workflows/main.yml (hunk 2 above)
Comment (sfc-gh-pbulawa) [reply .github/workflows/main.yml]: Fixed

File: Snowflake.Data.Tests/SFBaseTest.cs [hunk 5]
```diff
@@ -82,8 +102,10 @@ public class TestEnvironmentSetup
         private const string ConnectionStringFmt = "scheme={0};host={1};port={2};" + 
                                                    "account={3};role={4};db={5};warehouse={6};user={7};password={8};";
         
-        public static TestConfig TestConfig { get; private set; }
-        
+        public static TestConfig s_testConfig { get; private set; }
+
+        public static Dictionary<string, TimeSpan> s_testPerformance;
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply Snowflake.Data.Tests/SFBaseTest.cs]: Good idea. Done.

File: Snowflake.Data.Tests/SFBaseTest.cs (hunk 3 above)
Comment (sfc-gh-pbulawa) [reply Snowflake.Data.Tests/SFBaseTest.cs]: Sure, rewrote it.

File: Snowflake.Data.Tests/SFBaseTest.cs [hunk 6]
```diff
@@ -116,33 +138,69 @@ public void Setup()
...
-            TestConfig.database,
-            TestConfig.warehouse,
-            TestConfig.user,
-            TestConfig.password);
+        [OneTimeTearDown]
+        private void CreateTestTimeArtifact()
+        {
+            var os = "";
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply Snowflake.Data.Tests/SFBaseTest.cs]: Good idea. Done

**Review by sfc-gh-pfus — APPROVED** [review]

**Pull request merged**